### PR TITLE
[CARBONDATA-283] VT enhancement for data retention concurrent scenarios

### PR DESCRIPTION
**Pull request by ManoharVanam**
why raise this PR ? 
1 . Handling concurrency scenarios for retention feature
           Reading latest table status metadata before updating the table status file
2.  Added test cases for concurrent scenarios like load, delete segment by date/ id & clean files operations.

### CODE REVIEW

File: processing/src/main/java/org/apache/carbondata/processing/util/CarbonTableStatusUtil.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+    return date;
+  }
+
+  /**
+   * updates segment status and modificaton time details
+   *
+   * @param loadMetadata
+   */
+  public static void updateSegmentMetadataDetails(LoadMetadataDetails loadMetadata) {
```
Comment (jackylk) [inline processing/src/main/java/org/apache/carbondata/processing/util/CarbonTableStatusUtil.java]: Can you improve on the function name to depict the behavior of this function?

File: processing/src/main/java/org/apache/carbondata/processing/util/CarbonTableStatusUtil.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+  /**
+   * updates table status details using latest metadata
+   *
+   * @param oldMetadata
+   * @param newMetadata
+   * @return
+   */
+
+  public static List<LoadMetadataDetails> updateLatestTableStatusDetails(
```
Comment (jackylk) [inline processing/src/main/java/org/apache/carbondata/processing/util/CarbonTableStatusUtil.java]: I think these should not be utility functions, but should be member function of LoadMetadataDetails

File: processing/src/main/java/org/apache/carbondata/processing/util/CarbonTableStatusUtil.java (hunk 1 above)
Comment (ManoharVanam) [inline processing/src/main/java/org/apache/carbondata/processing/util/CarbonTableStatusUtil.java]: ok

File: processing/src/main/java/org/apache/carbondata/processing/util/CarbonTableStatusUtil.java (hunk 2 above)
Comment (ManoharVanam) [inline processing/src/main/java/org/apache/carbondata/processing/util/CarbonTableStatusUtil.java]: ok

**Comment by ManoharVanam** [comment]
http://136.243.101.176:8080/job/ApacheCarbonManualPRBuilder/574/

**Comment by gvramana** [comment]
LGTM

**Pull request closed without merging**